src/sagemaker_xgboost_container/serving.py

```diff
@@ -88,13 +97,26 @@ def default_output_fn(prediction, accept):
 
 
 def _user_module_transformer(user_module):
-    model_fn = getattr(user_module, 'model_fn', default_model_fn)
-    input_fn = getattr(user_module, 'input_fn', default_input_fn)
-    predict_fn = getattr(user_module, 'predict_fn', default_predict_fn)
-    output_fn = getattr(user_module, 'output_fn', default_output_fn)
-
-    return transformer.Transformer(model_fn=model_fn, input_fn=input_fn, predict_fn=predict_fn,
-                                   output_fn=output_fn)
+    model_fn = getattr(user_module, "model_fn", default_model_fn)
+    input_fn = getattr(user_module, "input_fn", None)
+    predict_fn = getattr(user_module, "predict_fn", None)
+    output_fn = getattr(user_module, "output_fn", None)
+    transform_fn = getattr(user_module, "transform_fn", None)
+
+    if transform_fn and (input_fn or predict_fn or output_fn):
```

Comment:
Transformer class is already throwing a [ValueError](https://github.com/aws/sagemaker-containers/blob/master/src/sagemaker_containers/_transformer.py#L154) for the same condition so curious if this would be redundant.

Reply:
The logic in this PR is structured so that the Transformer won't enter that if statement. Also, I thought it should be raised as a customer error on our side rather than a generic ValueError.